### PR TITLE
readme: add additional details

### PR DESCRIPTION
**Pull request by joostjager**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,65 @@
 # Lightning Multiplexer
 
-This repository contains the code for the `lnmux` service. The problem that `lnmux` solves is fail-over for incoming Lightning payments. BOLT11 invoices are bound to a single node. If that node goes down, the invoice is no longer payable. The same is true for the node or its peers running out of inbound liquidity.
+This repository contains the code for Lightning Multiplexer service, `lnmux` for short.
+
+The problem that `lnmux` solves is fail-over for incoming Lightning payments. BOLT11 invoices are bound to a single node. If that node goes down, the invoice is no longer payable. The same is true for the node or its peers running out of inbound liquidity.
+
+A typical solution is to set up multiple nodes. If invoice expiry durations are short and down time is planned, it is possible to gracefully spin down a node by waiting for all invoices generated by that node to reach a final state. During that time, new invoices should only be generated on the remaining nodes.
+
+This however does not work for unexpected down time. An invoice that may have been passed on to a third party already becomes impossible to settle.
```
Comment (aakselrod) [inline README.md]: nit: is this a third party, or the payer of the invoice (second party)?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,65 @@
...
 
 What `lnmux` allows you to do is set up a cluster of nodes where each of these nodes can settle any invoice. If a node is unable to accept payments for one of the reasons above, senders will still be able to pay the invoice via one of the other nodes.
 
 Even with all nodes in the cluster available, it can be advantageous for the sender to have multiple routing options to better utilize liquidity and minimise fees. If needed, multi-part payments can be completed through more than one cluster node.
 
+## Design
+
+`lnmux` is a massively stripped down version of a full Lightning node, almost to the point that there is nothing left. It has no channels, no external network connections and it does not partake in p2p gossip. It does have its own node key for signing invoices and it is also able to decode onion packets.
+
+The intended setup is a single instance of `lnmux` in conjunction with a number of full nodes.
```
Comment (aakselrod) [inline README.md:19]: nit: "full node" seems like it specifies bitcoin nodes rather than LN nodes

**Review by aakselrod — APPROVED** [review]
This is a great update! A couple of small nits, but consider them suggestions only.

File: README.md (hunk 1 above)
Comment (joostjager) [reply README.md]: Ah good point, this is written from the perspective of a custodian. Updated.

File: README.md (hunk 2 above)
Comment (joostjager) [reply README.md:19]: I couldn't come up with a better name for it, but added an explanatory sentence that in this context 'full node' means full lightning node.

Comment (aakselrod) [reply README.md:19]: Looks great!

File: README.md (hunk 1 above)
Comment (aakselrod) [reply README.md]: Ah, that makes sense. Looks good!

**Review by aakselrod — APPROVED** [review]

**Comment by Torakushi** [comment]
nice!

**Pull request merged**